File: llm_outputs/python_programs_corrected_by_llama/quicksort.py
def quicksort(arr):
    if len(arr) <= 1:
        return arr
    else:
        pivot = arr[0]
        lesser = quicksort([x for x in arr[1:] if x <= pivot])
        greater = quicksort([x for x in arr[1:] if x > pivot])
        return lesser + [pivot] + greater

File: llm_outputs/java_programs_corrected_by_llama/QUICKSORT.java
package java_programs;
import java.util.*;

public class QUICKSORT {
    public static ArrayList<Integer> quicksort(ArrayList<Integer> arr) {
        if (arr.isEmpty()) {
            return new ArrayList<>();
        }

        int pivot = arr.get(0);
        ArrayList<Integer> lesser = new ArrayList<>();
        ArrayList<Integer> greater = new ArrayList<>();

        for (int i = 1; i < arr.size(); i++) {
            if (arr.get(i) < pivot) {
                lesser.add(arr.get(i));
            } else if (arr.get(i) > pivot) {
                greater.add(arr.get(i));
            }
        }
        ArrayList<Integer> middle = new ArrayList<>(Collections.singletonList(pivot));
        middle.addAll(greater);
        lesser.addAll(middle);
        return lesser;
    }
}